confetti/src/main/java/com/github/jinatonic/confetti/confetto/Confetto.java

```diff
@@ -304,18 +315,22 @@ public boolean applyUpdate(long passedTime) {
         return !terminated;
     }
 
-    private float computeDistance(long t, float xi, float vi, float ai, Long targetTime,
-            Float vTarget) {
+    private Pair<Float, Float> computeDistance(long t, float xi, float vi, float ai, Long targetTime, Float vTarget) {
         if (targetTime == null || t < targetTime) {
-            // distance covered with linear acceleration
+            // velocity with constant acceleration
+            float vX = ai * t + vi;
+            // distance covered with constant acceleration
             // distance = xi + vi * t + 1/2 * a * t^2
-            return xi + vi * t + 0.5f * ai * t * t;
+            float x = xi + vi * t + 0.5f * ai * t * t;
+            return Pair.create(x, vX);
         } else {
-            // distance covered with linear acceleration + distance covered with max velocity
+            // velocity with constant acceleration
+            float vX = ai * t + vi;
+            // distance covered with constant acceleration + distance covered with max velocity
             // distance = xi + vi * targetTime + 1/2 * a * targetTime^2
             //     + (t - targetTime) * vTarget;
-            return xi + vi * targetTime + 0.5f * ai * targetTime * targetTime
-                    + (t - targetTime) * vTarget;
+            float x = xi + vi * targetTime + 0.5f * ai * targetTime * targetTime + (t - targetTime) * vTarget;
+            return Pair.create(x, vX);
```

Comment:
Creating a `Pair` object per `Confetto` per frame is fairly expensive, I would do some refactor around this function to avoid having to create this `Pair`.